library/source/wrapper/DdUrlTransfer.bs

```diff
@@ -0,0 +1,651 @@
+    end sub
+
+    ' ----------------------------------------------------------------
+    ' Returns the current URL.
+    '
+    ' @return (string) The Url
+    ' ----------------------------------------------------------------
+    function GetUrl() as string
+        return m.roUrlTransfer.GetUrl()
+    end function
+
+    ' ----------------------------------------------------------------
+    ' Changes the request method from the normal GET, HEAD or POST to the value passed as a string.
+    '
+    ' @param req (string) The request method to be used
+    ' ----------------------------------------------------------------
+    sub SetRequest(req as string)
+        m.roUrlTransfer.SetRequest(req)
+    end sub
+
+    ' ----------------------------------------------------------------
+    ' Returns the current request method.
+    '
+    ' @return (string) The request method
+    ' ----------------------------------------------------------------
+    function GetRequest() as string
+        return m.roUrlTransfer.GetRequest()
+    end function
+
+    ' ----------------------------------------------------------------
+    ' Connects to the remote service as specified in the URL and returns
+    ' the response body as a string. This function waits for the transfer
+    ' to complete and it may block for a long time. This calls discards
+    ' the headers and response codes. If that information is required,
+    ' use the AsyncGetToString() method.
+    '
+    ' @return (string) The response body
+    ' ----------------------------------------------------------------
+    function GetToString() as string
+        timer = CreateObject("roTimespan")
+        port = CreateObject("roMessagePort")
+        m.roUrlTransfer.SetMessagePort(port)
+
+        url = m.roUrlTransfer.GetUrl()
+        traceId = generateUniqueId()
+        spanId = generateUniqueId()
+        m.roUrlTransfer.AddHeader("x-datadog-trace-id", traceId)
+        m.roUrlTransfer.AddHeader("x-datadog-parent-id", spanId)
+        m.roUrlTransfer.AddHeader("x-datadog-sampling-priority", "1")
+        m.roUrlTransfer.AddHeader("x-datadog-origin", "rum")
+        timer.Mark()
+        result = m.roUrlTransfer.AsyncGetToString()
+
+        while (true)
+            msg = wait(5000, port)
+            if (msg <> invalid)
+                msgType = type(msg)
+                if (msgType = "roUrlEvent")
+                    if (msg.GetInt() = 1) ' transfer complete
+                        durationMs& = timer.TotalMilliseconds()
+                        transferTime# = millisToSec(durationMs&)
+                        response = msg.GetString()
+                        bytesDownloaded = Len(response)
+                        httpCode = msg.GetResponseCode()
+                        status = "ok"
+                        if (httpCode < 0)
+                            status = msg.GetFailureReason()
+                        end if
+                        resource = {
```

Comment:
it seems `rule_psr` property is missing (even if we don't do any sampling, we can add it).

Reply:
I have a later task to actually implement sampling so I'll add it there